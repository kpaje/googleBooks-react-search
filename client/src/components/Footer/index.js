import React from "react";

function Footer() {
	return (
		<footer>
			<hr />
			<p className="pull-right">
				<i className="fab fa-github" /> Kelvin Paje
			</p>
		</footer>
	);
}

export default Footer;
