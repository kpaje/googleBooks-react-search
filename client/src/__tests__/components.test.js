import React from "react";
import { create } from "react-test-renderer";
import Book from "../components/Book/index";
import Card from "../components/Card/index";
import Footer from "../components/Footer/index";
import Form from "../components/Form/index";
import Grid from "../components/Grid/index";
import Jumbotron from "../components/Jumbotron/index";
import List from "../components/List/index";
import Nav from "../components/Nav/index";

describe("Render Book component correctly", () => {
	test("Matches the snapshot", () => {
		const book = create(<Book />);
		expect(book.toJSON()).toMatchSnapshot();
	});
});

describe("Render Card component correctly", () => {
	test("Matches the snapshot", () => {
		const card = create(<Card />);
		expect(card.toJSON()).toMatchSnapshot();
	});
});

describe("Render Footer component correctly", () => {
	test("Matches the snapshot", () => {
		const footer = create(<Footer />);
		expect(footer.toJSON()).toMatchSnapshot();
	});
});

describe("Render Form component correctly", () => {
	test("Matches the snapshot", () => {
		const form = create(<Form />);
		expect(form.toJSON()).toMatchSnapshot();
	});
});

describe("Render Grid component correctly", () => {
	test("Matches the snapshot", () => {
		const grid = create(<Grid />);
		expect(grid.toJSON()).toMatchSnapshot();
	});
});

describe("Render Jumbotron component correctly", () => {
	test("Matches the snapshot", () => {
		const jumbotron = create(<Jumbotron />);
		expect(jumbotron.toJSON()).toMatchSnapshot();
	});
});

describe("Render List component correctly", () => {
	test("Matches the snapshot", () => {
		const list = create(<List />);
		expect(list.toJSON()).toMatchSnapshot();
	});
});

describe("Render Nav component correctly", () => {
	test("Matches the snapshot", () => {
		const nav = create(<Nav />);
		expect(nav.toJSON()).toMatchSnapshot();
	});
});
