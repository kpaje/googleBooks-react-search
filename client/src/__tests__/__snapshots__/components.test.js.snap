// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Card component correctly Matches the snapshot 1`] = `
<div
  className="card mt-4"
>
  <div
    className="card-header"
  >
    <h3>
      <strong>
        <i
          aria-hidden="true"
          className="fa fa-undefined"
        />
         
      </strong>
    </h3>
  </div>
  <div
    className="card-body"
  />
</div>
`;

exports[`Render Footer component correctly Matches the snapshot 1`] = `
<footer>
  <hr />
  <p
    className="pull-right"
  >
    <i
      className="fab fa-github"
    />
     Kelvin Paje
  </p>
</footer>
`;

exports[`Render Form component correctly Matches the snapshot 1`] = `
<form>
  <div
    className="form-group"
  >
    <label
      htmlFor="Query"
    >
      <strong>
        Book
      </strong>
    </label>
    <input
      className="form-control"
      id="Title"
      name="query"
      placeholder="Ready Player One"
      required={true}
      type="text"
    />
  </div>
  <div
    className="pull-right"
  >
    <button
      className="btn btn-lg btn-danger float-right"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;

exports[`Render Jumbotron component correctly Matches the snapshot 1`] = `
<div
  className="jumbotron mt-4"
/>
`;
